controllers/azuremachine_reconciler.go

```diff
@@ -222,6 +227,7 @@ func (s *azureMachineService) reconcileVirtualMachine(ctx context.Context, nicNa
 	vmSpec := &virtualmachines.Spec{
 		Name:                   s.machineScope.Name(),
 		NICName:                nicName,
```

Comment:
Would it be better to refactor it to an array of nics rather than primary and secondary? The code  in vms would be much simpler to just loop over the nics rather than primary and secondary

Reply:
yes, will change if this solution actually works for the Talos provider